addons/point_of_sale/models/res_config_settings.py

```diff
@@ -10,3 +11,11 @@ class ResConfigSettings(models.TransientModel):
     module_pos_mercury = fields.Boolean(string="Vantiv Payment Terminal", help="The transactions are processed by Vantiv. Set your Vantiv credentials on the related payment method.")
     module_pos_adyen = fields.Boolean(string="Adyen Payment Terminal", help="The transactions are processed by Adyen. Set your Adyen credentials on the related payment method.")
     update_stock_quantities = fields.Selection(related="company_id.point_of_sale_update_stock_quantities", readonly=False)
+
+    @api.onchange('group_product_pricelist')
+    def _check_pos_config_pricelist(self):
+        self.ensure_one()
```

Comment:
You should not write on records from a onchange.

For this task, what you probably should do is to overwrite the method 'def set_values(self):', you can have an example in the module stock.

And in this override, you can search pos config and change their usepricelist value depending on the group